### DISCUSSION TITLE
APQ support in DGS

### DISCUSSION THREAD

**Discussion by saravind-hotstar** [Netflix/dgs-framework#788] [Q&A]
Can APQ supported in DGS, currently most of the services are in DGS, we cannot enable APQ primarily due to limitation in DGS.
Please can you guide how to enable or implement it.

**Comment by saravind-hotstar** [comment]
@paulbakker @berngp Please could you let us know could we expect this feature, if not I could help building this feature with some guidance

**Comment by berngp** [comment]
@saravind-hotstar, sorry we haven’t provided an answer for this yet. We don't leverage APQ internally via DGS since our _Federation Gateway_ handles it directly. This will be a feature that, at least for now, would be mainly for OSS. All these said, we have consider APQ and will provide a mechanism to support it in the next couple of weeks. Will let you know, via this thread, when it is ready for review.

**Reply by saravind-hotstar** [reply]
@berngp Thanks a lot !

**Comment by saravind-hotstar** [comment]
@berngp Sorry to bother you, it would be of great help, if we can know the target milestone for this feature. This would help us planning dependencies bit easy. Thanks a lot

**Reply by berngp** [reply]
Working on it this week, will update as soon as I have an RC you can start using.

**Reply by berngp** [reply]
Created PR #829 with an initial implementation of this feature. The PR is up for review and cleanup but describes some ergonomics that you, as a user, will have available. Feedback is welcomed since at this stage is easy to influence the ergonomics the framework will provide you.
